src/error.rs

```diff
@@ -155,23 +147,16 @@ impl ConnectionError {
     // closes during a message receive loop, we could quietly end the stream instead). This method
     // helps to centralise that discrimination.
     pub(crate) fn is_benign(&self) -> bool {
-        match self {
+        matches!(
+            self,
             // We expect that timeouts will be used to naturally close connections once they are
             // idle, so we treat it as benign.
-            Self::TimedOut => true,
-
+            Self::TimedOut |
             // A graceful close from our end.
-            Self::Closed(Close::Local) => true,
-
-            // A graceful close from the peer, with the default code and empty reason.
-            Self::Closed(Close::Application {
-                error_code: 0,
-                reason,
-            }) if reason.is_empty() => true,
-
-            // Any other error is classified as not benign, so should always be propagated.
-            _ => false,
-        }
+            Self::Closed(Close::Local) |
+            // A graceful close from the peer, with the default code 0
+            Self::Closed(Close::Application { error_code: 0, .. })
```

Comment:
previously only `reason.is_empty()`  will be classified as benigh for Close::Application . 
now it is for as long as error_code is 0 ,
is it expected?